### PR TITLE
carousel is functioning, need to add black background and add other s…

### PR DESCRIPTION
**Pull request by nguyend08**
…tyles for less than 9 photos on page

### CODE REVIEW

File: client/src/components/app.jsx [hunk 1]
```diff
@@ -23,11 +24,15 @@ class App extends React.Component {
     };
     this.previousSlide = this.previousSlide.bind(this);
     this.nextSlide = this.nextSlide.bind(this);
-    this.handleClick = this.handleClick.bind(this);
+    this.handleCarouselClick = this.handleCarouselClick.bind(this);
+    this.handleHeaderClick = this.handleHeaderClick.bind(this);
+    this.handleExitClick = this.handleExitClick.bind(this);
   }
+  //'http://localhost:3005/restaurants/52566848/photos'
```
Comment (chrislarson) [inline client/src/components/app.jsx:31]: Remove commented code

File: client/src/components/app.jsx [hunk 2]
```diff
@@ -23,11 +24,15 @@ class App extends React.Component {
...
+  ///api/restaurants/${window.location.pathname}/photos
 
   componentDidMount() {
-    axios.get('http://localhost:3005/restaurants/52566848/photos').then((data) => {
+    axios.get(`http://localhost:3005/restaurants/52566848/photos`).then((data) => {
```
Comment (chrislarson) [inline client/src/components/app.jsx:35]: Remove hard-coding

File: server/index.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ app.use(bodyParser.json());
 
 app.get('/', (req, res) => res.send('WE ARE CONNECTED!!'));
 
+///api/restaurants/:restaurantId/photos
```
Comment (chrislarson) [inline server/index.js:22]: Remove commented code

File: client/src/components/style-9.css [hunk 4]
```diff
@@ -37,11 +37,44 @@
...
+}
+
+.footer-text {
+  margin: auto;
+  width: 590px;
+  padding-top: 10px;
+}
+
+.exit-button {
+  /* padding-bottom: 10px;
```
Comment (chrislarson) [inline client/src/components/style-9.css:55]: Remove commented code

File: client/src/components/app.jsx [hunk 5]
```diff
@@ -88,12 +107,19 @@ class App extends React.Component {
...
+
           <ImageSlide url={this.state.photos[this.state.currentImageIndex].url} />
 
           <Arrow
             direction="right"
             clickFunction={this.nextSlide}
             glyph="&#9654;" />
+
+          <Footer
+            publisher={this.state.photos[this.state.currentImageIndex].publisherName}
```
Comment (chrislarson) [inline client/src/components/app.jsx:120]: Destructuring at the top of the render function would make this easier to read.. you can do ``` const { photos, currentImageIndex } = this.state ``` and then use those variables directly :+1:

File: client/src/components/app.jsx [hunk 6]
```diff
@@ -72,10 +91,10 @@ class App extends React.Component {
         <PhotoBody>
           <div className="photo-gallery-body">
             <div>
-              <Header Photos={this.state.photos} />
+              <Header Photos={this.state.photos} handleHeaderClick={this.handleHeaderClick} />
             </div>
             <div>
-              <PhotoList Photos={this.state.photos} handleClick={this.handleClick} />
+              <PhotoList Photos={this.state.photos} handleCarouselClick={this.handleCarouselClick} />
```
Comment (chrislarson) [inline client/src/components/app.jsx:97]: If you do the destructuring noted in other comment, you can use the variable directly here too :+1:

File: client/src/components/photo-entry.jsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ class PhotoEntry extends React.Component {
   render() {
     return (
       <Photos photoNum={`photo${this.props.photo.id}`}>
-        <Image photoNum={`photo${this.props.photo.id}`} src={this.props.photo.url} alt="" onClick={() => this.props.handleClick()}/>
+        <Image photoNum={`photo${this.props.photo.id}`} src={this.props.photo.url} alt="" onClick={() => this.props.handleClick(this.props.photo.id)}/>
```
Comment (chrislarson) [inline client/src/components/photo-entry.jsx:13]: Recommend destructuring

File: client/src/components/app.jsx [hunk 8]
```diff
@@ -72,10 +91,10 @@ class App extends React.Component {
         <PhotoBody>
           <div className="photo-gallery-body">
             <div>
-              <Header Photos={this.state.photos} />
+              <Header Photos={this.state.photos} handleHeaderClick={this.handleHeaderClick} />
             </div>
```
Comment (chrislarson) [inline client/src/components/app.jsx:95]: It seems like there are a lot of extra divs in your render functions; if these components (Header for example) return divs, there is no reason to wrap them in an extra div.. the extra divs make it a little harder to see what's happening in the code and in the dev tools element panel

File: client/src/components/app.jsx [hunk 9]
```diff
@@ -88,12 +107,19 @@ class App extends React.Component {
             clickFunction={this.previousSlide}
             glyph="&#9664;" />
 
+          <div className="exit-button" onClick={() => this.handleExitClick()}><span>X</span></div>
```
Comment (GaoR) [inline client/src/components/app.jsx:110]: No need for anonymous function wrapper when function is already bound in the constructor.  DRY!

**Review by GaoR — APPROVED** [review]
Good work, be sure to install ES Lint though!

**Pull request merged**